internal/gcp/tags.go

```diff
@@ -150,3 +153,16 @@ func cacheKeyTagKey(key string) string {
 func cacheKeyTagValue(key string, value string) string {
 	return fmt.Sprintf("value:%s:%s", key, value)
 }
+
+func (m *tagsManager) GetProjectInfo(ctx context.Context, projectID string) (*resourcemanagerpb.Project, error) {
+
+	req := &resourcemanagerpb.GetProjectRequest{
+		Name: "projects/" + projectID,
+	}
+
+	project, err := m.projectsClient.GetProject(ctx, req)
```

Comment:
Might be worthwhile to cache this information too, since we already have the cache client anyway. We should be able to use a longer cache duration here too.